_resources/port1.0/group/cmake-1.1.tcl

```diff
@@ -194,6 +254,7 @@ pre-configure {
         ui_debug "Adding -DCMAKE_INSTALL_RPATH=[join ${cmake.install_rpath} \;] to configure.args"
         configure.args-append -DCMAKE_INSTALL_RPATH="[join ${cmake.install_rpath} \;]"
     }
+    configure.pre_args-prepend "-G \"[join ${cmake.generator}]\""
```

Comment:
I think you'll have to keep the join anyway (due to `cmake.generator` always containing a list), but since it'll be a single-element list at all times, it shouldn't really make a difference (i.e. always return the same as `[lindex ${cmake.generator} 0]` (and join is much more readable then).